projects/packages/partner/src/class-partner-coupon.php

```diff
@@ -127,27 +153,123 @@ public function catch_coupon( $plugin_slug, $redirect_location ) {
 	/**
 	 * Purge partner coupon.
 	 *
-	 * We automatically purge partner coupons after a certain amount of time to prevent
-	 * us from unnecessarily promoting a product for months or years in the future.
+	 * We try to remotely check if a coupon looks valid. We also automatically purge
+	 * partner coupons after a certain amount of time to prevent unnecessary look-ups
+	 * and/or promoting a product for months or years in the future due to unknown
+	 * errors.
+	 */
+	public function maybe_purge_coupon() {
+		// Only run coupon checks on Jetpack admin pages.
+		// The "admin-ui" package is responsible for registering the Jetpack admin
+		// page for all Jetpack plugins and has hardcoded the settings page to be
+		// "jetpack", so we shouldn't need to allow for dynamic/custom values.
+		// phpcs:ignore WordPress.Security.NonceVerification.Recommended
+		if ( ! isset( $_GET['page'] ) || 'jetpack' !== $_GET['page'] ) {
+			return;
+		}
+
+		if ( $this->maybe_purge_coupon_by_added_date() ) {
+			return;
+		}
+
+		// Limit checks to happen every minute instead of on all page requests.
```

Comment:
I'd reword this to "Limit checks to happen once a minute at most." otherwise this sounds like it executes every minute like a cron job.